movement/utils/dimensions.py

```diff
@@ -0,0 +1,113 @@
+"""Utilities for manipulating dimensions of ``xarray.DataArray`` objects."""
+
+from collections.abc import Hashable, Iterable
+
+import xarray as xr
+
+
+def collapse_extra_dimensions(
+    da: xr.DataArray,
+    preserve_dims: Iterable[str] = ("time", "space"),
+    **selection: int | str,
+) -> xr.DataArray:
+    """Collapse a ``DataArray``, preserving only the specified dimensions.
+
+    By default, dimensions that are collapsed retain the corresponding 'slice'
+    along their 0th index of those dimensions, unless a particular index for is
+    given in the ``selection``.
+
+    Parameters
+    ----------
+    da : xarray.DataArray
+        DataArray of multiple dimensions, which is to be collapsed.
+    preserve_dims : Iterable[str]
+        The dimensions of ``da`` that should not be collapsed.
+    selection : int | str
+        Mapping from dimension names to a particular index in that dimension.
+        Dimensions that appear with an index in ``selection`` retain that index
+        slice when collapsed, rather than the default 0th index slice.
+
+    Returns
+    -------
+    xarray.DataArray
+        DataArray whose shape is the same as the shape of the preserved
+        dimensions of ``da``, containing the data obtained from a slice along
+        the collapsed dimensions.
+
+    Examples
+    --------
+    Collapse a ``DataArray`` down to just a ``"time"``-``"space"`` slice.
+
+    >>> import xarray as xr
+    >>> import numpy as np
+    >>> shape = (7, 2, 3, 4)
+    >>> da = xr.DataArray(
+    ...     data=np.arange(np.prod(shape)).reshape(shape),
+    ...     dims=["time", "space", "dim_to_collapse_0", "dim_to_collapse_1"],
+    ... )
+    >>> space_time = collapse_extra_dimensions(da)
+    >>> print(space_time.shape)
+    (7, 2)
+
+    The call to ``collapse_extra_dimensions`` above is equivalent to
```

Comment:
This is equivalence statement is wrong, because you are using `.sel` to index by integer. See xarray's [quick overview](https://tutorial.xarray.dev/fundamentals/02.1_indexing_Basic.html#quick-overview) of indexing methods.

The correct statement would be:

```rst
  The call to ``collapse_extra_dimensions`` above is equivalent to
  ``da.isel(keypoints=0, individuals=0)`` (indexing by integer) and to
  ``da.sel(keypoints="head", individuals="Alice")`` (indexing by label).
  We can change which slice we take from the collapsed dimensions by passing
  them as keyword arguments.
```

Reply:
The reason it all works out in the tests is because coordinate indexes (integers) are converted to coordinate names by `coord_of_dimension`,  and then only the names (not the integer indices) are passed to .sel during assertions. 

TLDR, tests are fine, but the examples here are wrong.